extensions/github/src/links.ts

```diff
@@ -125,11 +125,11 @@ export function getPermalink(gitAPI: GitAPI, useSelection: boolean, hostPrefix?:
 		return;
 	}
 
-	const commitHash = (gitRepo.state.HEAD?.ahead === 0) ? `/blob/${gitRepo.state.HEAD?.commit}` : '';
+	const blobSegment = (gitRepo.state.HEAD?.ahead === 0) ? `/blob/${linkType === 'headlink' ? gitRepo.state.HEAD.name : gitRepo.state.HEAD?.commit}` : '';
```

Comment:
Don't we need to url encode the name? from what I know branch names can contain special characters.

Reply:
+

Reply:
This doesn't seem required elsewhere, e.g. https://insiders.vscode.dev/github.com/microsoft/vscode-pull-request-github/blob/57dd87743f1caa943ef5e65bfc0d6dcf101a0ee6/src/issues/util.ts#L559-L576

Reply:
Sure, but thats an issue isn't it.